### PR TITLE
add get all articles endpoint, controller, model and tests

### PR DESCRIPTION
**Pull request by ChrisRistoff**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -55,3 +55,25 @@ describe('topics', () => {
...
+        "article_id": expect.any(Number),
+        "topic": expect.any(String),
+        "created_at": expect.any(String),
+        "votes": expect.any(Number),
+        "article_img_url": expect.any(String),
+        "comment_count": expect.any(Number),
+      }))
+    }
+  })
+})
```
Comment (saimakhanom) [inline __tests__/integration.test.js]: you should be testing to see if the articles are sorted in the correct order

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -55,3 +56,28 @@ describe('topics', () => {
...
 })
+
+describe('articles', () => {
+  it('GET 200: Shoulrd return an array of objects containing articles to the user', async() => {
+    const res = await supertest(app).get("/api/articles")
+
+    expect(res.statusCode).toBe(200)
+    expect(Array.isArray(res.body.articles)).toBe(true)
+
+    expect(res.body.articles).toBeSortedBy("created_at", {descending: true})
```
Comment (nc-hev-mag) [inline __tests__/integration.test.js:66]: Brill!

File: models/articlesModel.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const articles = await db.query(`
+    SELECT
+      a.author,
+      a.title,
+      a.article_id,
+      a.topic,
+      a.created_at,
+      a.votes,
+      a.article_img_url,
+      CAST(COUNT(c.comment_id) AS INTEGER) AS comment_count
```
Comment (nc-hev-mag) [inline models/articlesModel.js:13]: Great SQL 💯

**Review by nc-hev-mag — APPROVED** [review]
Nice one Krasen, good to merge 🎉

**Pull request merged**